runtime/caml/signals.h

```diff
@@ -30,8 +30,17 @@
 extern "C" {
 #endif
 
-CAMLextern void caml_enter_blocking_section (void);
-CAMLextern void caml_leave_blocking_section (void);
+CAMLextern void caml_enter_blocking_section (void); // raises
+CAMLextern void caml_leave_blocking_section (void); // raises
+
+CAMLextern value caml_enter_blocking_section_exn (void);
+/* Same as [caml_enter_blocking_section], but return the exception if
+   any. The runtime lock is held upon return if and only if the result
+   satisfies Is_exception_result. */
+
+CAMLextern value caml_leave_blocking_section_exn (void);
+/* Same as [caml_leave_blocking_section], but return the exception if
+   any. The runtime lock is held upon return in all cases. */
```

Comment:
Thanks for the review, I just notice it now.

Indeed, I will rebase on top of #9722 and follow whatever decision is made there. However, I mentioned at #9722 a serious breakage of not polling in `caml_leave_blocking_section` that I just noticed: it serves to poll after a system call returns EINTR, just before `Sys_error`/`Unix_error(EINTR,_)` is raised. Once the exception is raised (changing a function raising `Sys.Break` to a function raising `Sys_error`, say), the argument that one will poll soon enough no longer applies for reasons I explained there. And since it is possible that this pattern could be found in user C libraries, fixing the Sys and Unix modules by hand would not be enough. I propose to discuss this at #9722.